### PR TITLE
feat: persona pre detemination

### PR DESCRIPTION
**Pull request by Lanttcat**
## Description

Closes #MF-1637 #MF-1851

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: packages/mask/src/components/shared/ApplicationBoard.tsx [hunk 1]
```diff
@@ -301,53 +273,32 @@ const ApplicationEntryStatusContext = createContext<ApplicationEntryStatusContex
...
     return (
         <ApplicationEntryStatusContext.Provider
             value={{
-                personaConnectAction: personaConnectStatus.action ?? undefined,
-                personaNextIDReset,
+                personaAction: personaConnectStatus.action,
                 isPersonaCreated: personaConnectStatus.hasPersona,
                 isPersonaConnected: personaConnectStatus.connected,
-                isNextIDVerify: nextIDConnectStatus.isVerified,
+                isNextIDVerify: personaConnectStatus.verified,
```
Comment (guanbinrui) [inline packages/mask/src/components/shared/ApplicationBoard.tsx:294]: What's the difference with `isPersonaConnected`?

File: packages/shared-base/src/NextID/index.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import { NextIDPlatform, ProfileIdentifier } from '../index.js'
+import { EnhanceableSite } from '../Site/type.js'
+
+export function nextIDPlatformToNetwork(platform: NextIDPlatform) {
```
Comment (guanbinrui) [inline packages/shared-base/src/NextID/index.ts:4]: We should not name this as `index.ts`.

File: packages/shared-base/src/NextID/index.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    // inline the pairs for testing, if out the pairs will not work
+    const pairs = [[NextIDPlatform.Twitter, EnhanceableSite.Twitter]]
+    const pair = pairs.find((x) => x[0] === platform)
+    return pair?.[1]
+}
+
+export function networkToNextIDPlatform(site: EnhanceableSite) {
```
Comment (guanbinrui) [inline packages/shared-base/src/NextID/index.ts:11]: Please use `createLookupTableResolver` stead.

File: packages/shared-base/src/NextID/index.ts (hunk 2 above)
Comment (guanbinrui) [inline packages/shared-base/src/NextID/index.ts:4]: Please use `createLookupTableResolver` stead.

File: packages/shared-base/src/NextID/index.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
+import { NextIDPlatform, ProfileIdentifier } from '../index.js'
```
Comment (guanbinrui) [inline packages/shared-base/src/NextID/index.ts:1]: We should not name this file as `index.ts`.

File: packages/mask/src/components/shared/ApplicationBoard.tsx (hunk 1 above)
Comment (Lanttcat) [reply packages/mask/src/components/shared/ApplicationBoard.tsx:294]: `Verified` mean  bound on nextID.
`Connected` mean on local.

File: packages/shared-base/src/NextID/index.ts (hunk 4 above)
Comment (Lanttcat) [reply packages/shared-base/src/NextID/index.ts:1]: Maybe reason is this:
```json
{
  "name": "@masknet/shared-base",
  "description": "This package MUST be pure (have no side effects when imported).",
  "type": "module"
}
```

File: packages/shared-base/src/NextID/index.ts (hunk 3 above)
Comment (Lanttcat) [reply packages/shared-base/src/NextID/index.ts:11]: I refactor in this pr: #7229

File: packages/mask/src/components/shared/PersonaBoundary.tsx [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+interface PersonaBoundaryProps {
+    handlerPosition?: 'center' | 'top-right'
+    directTo?: PluginId
+    customHint?: boolean
+    children: SupportChildren
+    enableVerify?: boolean
+    beforeVerify?: () => void | Promise<void>
+}
+
+export const PersonaBoundary = memo<PersonaBoundaryProps>(
```
Comment (guanbinrui) [inline packages/mask/src/components/shared/PersonaBoundary.tsx]: `PersonaBoundary` is not specific enough. Some candidates for replacing it:

+ `ConnectPersonaBoundary`
+ `ConnectNextIDBoundary`

File: packages/mask/src/components/shared/PluginEnableBoundary.tsx [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+import { useAsyncFn } from 'react-use'
+
+const useStyles = makeStyles()((theme) => ({
+    button: {
+        borderRadius: '99px',
+        backgroundColor: theme.palette.maskColor.dark,
+        color: '#fff',
+        marginTop: 'auto',
+        ':hover': {
+            color: 'fff',
```
Comment (guanbinrui) [inline packages/mask/src/components/shared/PluginEnableBoundary.tsx]: ?

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**